### PR TITLE
Reimplement notification groups with tf-framework

### PR DESCRIPTION
**Pull request by tosuke**
Output from acceptance testing:

```
$ MACKEREL_EXPERIMENTAL_TFFRAMEWORK=1 make testacc TESTS='"TestAcc(DataSource)?MackerelNotificationGroup"'
TF_ACC=1 go test -v ./mackerel/... -run "TestAcc(DataSource)?MackerelNotificationGroup" -timeout 120m
2024/07/03 20:27:09 [INFO] mackerel: use terraform-plugin-framework based implementation
=== RUN   TestAccDataSourceMackerelNotificationGroup
=== PAUSE TestAccDataSourceMackerelNotificationGroup
=== RUN   TestAccMackerelNotificationGroup
=== PAUSE TestAccMackerelNotificationGroup
=== CONT  TestAccDataSourceMackerelNotificationGroup
=== CONT  TestAccMackerelNotificationGroup
--- PASS: TestAccDataSourceMackerelNotificationGroup (6.15s)
--- PASS: TestAccMackerelNotificationGroup (10.66s)
PASS
ok      github.com/mackerelio-labs/terraform-provider-mackerel/mackerel 12.024s
```

### CODE REVIEW

File: internal/provider/data_source_mackerel_notification_group.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+							Description: "the name of the service",
+
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_notification_group.go:97]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/3c50804cc9e136fe2e9a4e6ec34815dea8cd75ef/mackerel/data_source_mackerel_notification_group.go?plain=1#L15-L48

File: internal/provider/resource_mackerel_notification_group.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+							Validators: []validator.String{
+								mackerel.ServiceNameValidator(),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_notification_group.go:117]: Original schema: https://github.com/tosuke/terraform-provider-mackerel/blob/3c50804cc9e136fe2e9a4e6ec34815dea8cd75ef/mackerel/resource_mackerel_notification_group.go?plain=1#L44-L75

File: internal/provider/resource_mackerel_notification_group.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
+							Description: "The name of the service",
+
+							Required: true,
+							Validators: []validator.String{
+								mackerel.ServiceNameValidator(),
+							},
+						},
+					},
```
Comment (tosuke) [inline internal/provider/resource_mackerel_notification_group.go:112]: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/3c50804cc9e136fe2e9a4e6ec34815dea8cd75ef/mackerel/resource_mackerel_notification_group.go?plain=1#L12-L19

File: internal/provider/resource_mackerel_notification_group.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+							Required: true,
+						},
+						"skip_default": schema.BoolAttribute{
+							Description: "If true, send notifications to this notification group only.",
+
+							Optional: true,
+							Computed: true,
+							Default:  booldefault.StaticBool(false),
+						},
+					},
```
Comment (tosuke) [inline internal/provider/resource_mackerel_notification_group.go:96]: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/3c50804cc9e136fe2e9a4e6ec34815dea8cd75ef/mackerel/resource_mackerel_notification_group.go?plain=1#L21-L33

File: internal/provider/resource_mackerel_notification_group.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+			},
+			"child_channel_ids": schema.SetAttribute{
+				Description: "A set of notification channel IDs",
+
+				ElementType: types.StringType,
+				Optional:    true,
+			},
+		},
+		// TODO: migrate to nested attributes
+		Blocks: map[string]schema.Block{
```
Comment (tosuke) [inline internal/provider/resource_mackerel_notification_group.go:78]: Blocks are a restricted version of nested object attributes (e.g. cannot be used as expressions).

File: internal/provider/resource_mackerel_notification_group.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+							Description: "The monitor rule ID",
+
+							Required: true,
+						},
+						"skip_default": schema.BoolAttribute{
+							Description: "If true, send notifications to this notification group only.",
+
+							Optional: true,
+							Computed: true,
+							Default:  booldefault.StaticBool(false),
```
Comment (tosuke) [inline internal/provider/resource_mackerel_notification_group.go:94]: Attributes with default value are both inputs and outputs. If the user provides a value, they are act as input. If not, the provider computes and provides a default value. So they need not to be only optional but also computed.

**Review by Arthur1 — APPROVED** [review]
:+1:

**Pull request merged**